pkg/bpf/prog.go

```diff
@@ -67,3 +74,72 @@ func (t ProgType) String() string {
 
 	return "Unknown"
 }
+
+type attrProg struct {
+	startID   uint32
+	progID    uint32
```

Comment:
In the uapi header the {start_id, prog_id, map_id} is a union, wouldn't this give wrong offset in GetNextID() and GetFDByID() as it currently is? Maybe this could just be named currID or objID as a single member here.

Reply:
added, and added a comment much like the one you have here: https://github.com/cilium/cilium/blob/master/pkg/bpf/bpf.go#L120 :)